src/services/AsyncStorageService.ts

```diff
@@ -1,26 +1,39 @@
 import AsyncStorage from '@react-native-async-storage/async-storage'
+import { OS_TYPES } from '../constants'
+import { Platform } from 'react-native'
 
 class AsyncStorageService {
   async getItem(key: string): Promise<unknown> {
     const jsonValue = await AsyncStorage.getItem(key)
-    const value = JSON.parse(jsonValue as string)
-    if (value) {
-      return value
+    try {
+      const value = JSON.parse(jsonValue as string)
+      if (value) {
+        return value
+      }
+    } catch (error) {
+      console.log('Error parsing JSON string: ', error)
+      if (jsonValue) {
+        return jsonValue
+      }
     }
 
     return null
   }
 
   async setItem(key: string, value: string): Promise<void> {
-    await AsyncStorage.setItem(key, JSON.stringify(value))
+    await AsyncStorage.setItem(key, value)
   }
 
   async removeItem(key: string): Promise<void> {
     await AsyncStorage.removeItem(key)
   }
 
   async clear(): Promise<void> {
-    await AsyncStorage.clear()
+    if (Platform.OS === OS_TYPES.ANDROID) {
+      await AsyncStorage.clear()
+    } else {
+      await AsyncStorage.getAllKeys().then(AsyncStorage.multiRemove)
+    }
```

Comment:
I thought `clear` works the same on both platforms. 
If there is any reason we are doing this please make sure to leave a comment (maybe even a GitHub issue link related to this)

Reply:
Yep, basically i had problems with iOS every time it would try and clear the `AsyncStorage`, came across this answer https://stackoverflow.com/questions/46736268/react-native-asyncstorage-clear-is-failing-on-ios, its working fine now

Reply:
Just leave the comment in the code with the link you already shared